config/pmd/pmd-ruleset.xml

```diff
@@ -11,9 +11,14 @@
 
   <rule ref="category/java/bestpractices.xml">
     <exclude name="AbstractClassWithoutAbstractMethod" />
+    <exclude name="JUnitAssertionsShouldIncludeMessage" />
+    <exclude name="JUnitTestContainsTooManyAsserts" />
   </rule>
 
   <rule ref="category/java/codestyle.xml">
+    <exclude name="LinguisticNaming"/>
```

Comment:
Let's remove this exclusion or configure it to your needs

Reply:
OK this is an interesting one. Fluent API's, especially those for testing frameworks generally break LinguisticNaming all over the place. The specific ones I struggled with were in the InProcessTestUtil class. In order for the methods to read properly from where they are used, the methods were defined for example like this:

   public static ArgumentMatcher<AbstractServerStatus> hasServiceStatus(ServiceStatus serviceStatus) {
        return s -> s != null && s.getServiceStatus() == serviceStatus;
    }

The problem is I'm not entirely sure LinguisticNaming is still applicable to Fluent API's. Sure we can modify it and remove all the checks for boolean method and variable prefixes, but there won't be much left.
My point is that LinguisticNaming adheres to pre-fluent rules. I'm sure certain projects won't implement fluent API's. This is probably a developer choice. But for the K8S Operator project I think this rule will constrain developers in implementing fluent methods

Reply:
@Kerruba  and @werbth  I had a  look at the configuration of this rule. I can think of scenarios where each of those settings except one or two will constrain fluent naming conventions. I would really prefer for this rule to be an optional inclusion.

Reply:
Ok I understand why you're asking for this and I think it could make sense. 
How do you feel about Suppressing the warning for the specific methods (or class) instead of excluding the entire rule?

Reply:
Its OK I have renamed all the violating fluent methods with the traditional non-fluent, LinguisticNaming prefixes. We can leave this rule.